### ISSUE TITLE
Is there an example configuration with VELOCITY_CONFIG = "file"?

### ISSUE THREAD

**Issue by wrongkindofdoctor** [NOAA-GFDL/MOM6#1024]
I am implementing the new FMS-IO calls in the [MOM_state_initialization.F90](https://github.com/NOAA-GFDL/MOM6/blob/dev/gfdl/src/initialization/MOM_state_initialization.F90) module. The [initialize_velocity_from_file](https://github.com/NOAA-GFDL/MOM6/blob/83d8297724b588e4ce193cd68a6d6c44b04f347c/src/initialization/MOM_state_initialization.F90#L1254) subroutine calls [MOM_read_vector](https://github.com/NOAA-GFDL/MOM6/blob/83d8297724b588e4ce193cd68a6d6c44b04f347c/src/framework/MOM_io.F90#L953) that wraps the old IO read routines, and performs some scaling on u and v components read in from a file. I want to replace the read calls with the new domain-decomposed read routines, which requires registering the u and v axes to obtain the correct decomposition. 

To ensure that I am looking for the correct axis names (e.g., 'x', 'y'), I need an input file that is used in cases where VELOCITY_CONFIG = "file" in MOM_parameter_doc.all file. I have been testing the new code with ocean-only/global_ALE_z, which does not use these routines (i.e., VELOCITY_CONFIG = "zero" in [MOM_parameter_doc.all](https://github.com/NOAA-GFDL/MOM6-examples/blob/018520acd3eb3d2f5766de3925ba89f05432597e/ocean_only/global_ALE/z/MOM_parameter_doc.all#L498)), and cannot locate a parameter file with this setting in the MOM6-examples. 

Can a MOM6 developer either point me toward a case I've overlooked, or provide an xml with a configuration that uses  VELOCITY_CONFIG = "file", and contains the path to the input file? Or, if this feature is not used, please let me know if your would like me to delete it.

**Comment by MJHarrison-GFDL** [comment]
We should keep this option, since it may be in use beyond MOM6-examples. It is of limited utility, since unlike the tracers which can exist on a different grid from the model, the velocity initialization needs to be done from the same grid, so it is more like a restart. 

Recommend a short run of the global_ALE case in order to create an initial condition file and set "VELOCITY_FILE" to point to the restart file in a subsequent run.

**Comment by wrongkindofdoctor** [comment]
Thanks, @MJHarrison-GFDL. Using a restart file means that the expected input for the _u_ and _v_ axes will be _lath_, _lonh_, _latq_, or _lonq_. I will include queries for these diagnostic axes associated with _u_ and _v_ to determine whether the axes are oriented in the 'x' or 'y' direction as required by the new axis registration routines.

**Comment by MJHarrison-GFDL** [comment]
@wrongkindofdoctor .  We do have a cartesian attribute function in FMS that checks for obvious names like 'lon','longitude' and units like 'degrees_E'. This is not entirely consistent with community metadata standards, e.g. we're not checking for  "standard_name". I recommend that we rely on the cartesian function  or something similar rather than relying on specific axis names.

**Comment by wrongkindofdoctor** [comment]
I added checks for units of "degrees_east", or "degrees_north" to determine whether an axis is oriented in the x- or y-direction, since [CF-compliant files contain "units" attributes for the coordinate data](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#latitude-coordinate). The FMS cartesian coordinate attributes are a GFDL-specific requirement, and do not permit domain-decomposed reads of CF-compliant files from external sources without modifying the attributes beforehand, which is particularly cumbersome for users with a large number of files.  

I know that the units may also be something like "degree_E", but of the files I've encountered in the MOM tests for domain-decomposed reads have these lat/lon units. If you would like to have an option for the user to specify the coordinate axes names, with the default assumption of "degrees_east" and "degrees_north", I can add that. The new IO is designed with the expectation that the user will supply all information necesssary to read/write the desired data, so I would like to avoid parsing strings for every potential attribute iteration that I can think of.
